cli/src/client/queries.rs

```diff
@@ -0,0 +1,73 @@
+use crate::client::Client;
+use graphql_client::*;
+use std::error::Error;
+
+// We have to declare types for scalars in the schema.
+type GraphQLDocument = String;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    query_path = "graphql/get_schema_query.graphql",
+    schema_path = "graphql/agm.graphql",
+    response_derives = "PartialEq, Debug",
+    deprecated = "warn"
+)]
+pub struct GetSchemaQuery;
+
+pub fn get_schema_query(
+    client: &Client,
+    variables: get_schema_query::Variables,
+) -> Result<Option<GraphQLDocument>, Box<dyn Error>> {
+    let response_body: Response<get_schema_query::ResponseData> =
+        client.send::<GetSchemaQuery>(variables)?;
+
+    let hash: Option<GraphQLDocument> = response_body.data.and_then(|data| {
+        data.service
+            .and_then(|service| service.schema.map(|schema| schema.hash))
+    });
+
+    Ok(hash)
```

Comment:
is the hash really a `GraphQLDocument`?

Reply:
oh I see, this is just an example

Reply:
Actually you found an issue, I initially tested with getting the entire schema which is a `GraphQLDocument` scalar, which is typedef'd into a String, hash happens to be a `String` too so this compiled. I'll fix that.